sdcm/ec2_client.py

```diff
@@ -122,7 +123,7 @@ def _request_spot_fleet(self, instance_type, image_id, region_name, network_if,
         if key_pair:
             fleet_config['LaunchSpecifications'][0].update({'KeyName': key_pair})
         if user_data:
-            fleet_config['LaunchSpecifications'][0].update({'UserData': self._encode_user_data(user_data)})
```

Comment:
what's the idea of removing our own user data ? all the rest of the things we put into it, won't work.

Reply:
@fruch 2 points here:
1. Do we actually make any use of it though? We pass an encoded json, which EC2 `cloud_init` doesn't read (at least in artifact tests) and dumps into `user-data.txt` in `/var/lib/cloud/instances/<instance_name>`. I don't see SCT reading that file or retrieving that data using the AWS API. I've looked in the code for how we're using this data and apart from passing it in at provision time, I don't really see us using it. Am I missing something obvious?
2. If we don't replace it, we can't use `cloud-init`, since we need to pass valid `cloud-init` code in `UserData` and we're taking up that space with our custom json.

Reply:
@KnifeyMoloko User data is used outside of the SCT repo. Look at the https://github.com/scylladb/scylla-machine-image project.
And it's files:
- https://github.com/scylladb/scylla-machine-image/blob/next/common/scylla_configure.py#L85
- https://github.com/scylladb/scylla-machine-image/blob/next/common/scylla_post_start.py#L34

@fruch is right that other stuff will be broken in such a case. So, need to have a solution that won't break anything.